javatests/arcs/core/storage/LocalStorageEndpointManagerTest.kt

```diff
@@ -0,0 +1,102 @@
+package arcs.core.storage
+
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtOperationAtTime
+import arcs.core.data.EntityType
+import arcs.core.data.Schema
+import arcs.core.data.SchemaFields
+import arcs.core.data.SchemaName
+import arcs.core.data.SingletonType
+import arcs.core.storage.keys.RamDiskStorageKey
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.core.storage.testutil.testStorageEndpointManager
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(JUnit4::class)
+class LocalStorageEndpointManagerTest {
+
+  @Test
+  fun manager_get_createsNewStore() = runTest {
+    val endpointManager = testStorageEndpointManager(this)
+
+    val firstEndpoint = endpointManager.get(storageOptionsFrom(), DUMMY_CALLBACK)
+
+    val secondEndpoint = endpointManager.get(storageOptionsFrom("newKey"), DUMMY_CALLBACK)
+
+    assertThat((firstEndpoint as LocalStorageEndpoint<*, *, *>).storeForTests)
```

Comment:
It might be nice to define a `StorageEndpoint.storeFortests` helper method at the bottom to reduce noise.

Reply:
I like that.